loadgen/task-create-vault.js

```diff
@@ -1,10 +1,7 @@
+/* global __dirname */
+import path from 'path';
 import { E } from '@agoric/eventual-send';
-import { amountMath } from '@agoric/ertp';
-//import { allComparable } from '@agoric/same-structure';
-import { pursePetnames, issuerPetnames } from './petnames';
-import { multiplyBy, makeRatio } from '@agoric/zoe/src/contractSupport';
 import { disp } from './display';
-import { allValues } from './allValues';
 
 // Prepare to create and close a vault on each cycle. We measure our
 // available BLD at startup. On each cycle, we deposit 1% of that value as
```

Comment:
We should be getting back all of our BLD if the loan is paid back. I think the BLD values should not be going down due to creating and repaying the loan, although the BLD values will go down from the AMM task. Are we not getting everything back?

Reply:
I haven't looked closely enough to check. I wanted to guard against losses due to the cycle getting interrupted in the middle (leaking the vault), or concurrent tasks drawing against the same BLD purse. Also I wasn't sure I understood the fee model: we deduct RUN but never BLD?

Reply:
We never deduct BLD as a fee in a vault in the Treasury. You get all collateral back if you close the vault. There is a chance that the collateral was liquidated though, so it is good to double check that you do get everything back. https://github.com/Agoric/agoric-sdk/blob/c2fed992254dd8a2557b0a7a08a309fb5f0fc536/packages/treasury/src/vault.js#L159